### ISSUE TITLE
[Feature Request] Hook to extend cron schedules

### ISSUE THREAD

**Issue by DanielHeath** [bensheldon/good_job#1050]
Context:

I have a job that should run every 17 hours. 

If I specify `every 17 hours`, the job actually runs at 5pm every day, because fugit interprets it as a cron string of `0 */17 * * *`.

I'd argue that's somewhat a bug in fugit (it should raise an error if given an 'every X hours' greater than 12, since it can't represent it).

I plan to work around this for now by subclassing `CronEntry` to override `next_at` and `original`, and then overriding `GoodJob.configuration.cron_entries` to return instances of my subclass for the relevant jobs.

Suggestions:

`CronEntry#fugit` is only used within `CronEntry` and is an implementation detail - could be made private.

Extending the cron configuration syntax to take a block and a description parameter, eg:

```
config.good_job.cron = { example: { 
  description: 'every 17 hours', 
  next_time: -> (last_run_at) { (last_run_at || Time.now) + 17.hours}, 
  class: 'ExampleJob'  
} }
```

**Comment by DanielHeath** [comment]
I'd be happy to write a PR for this feature, if it would be accepted.

**Comment by bensheldon** [comment]
I really like this! I think my preference would be to overload the existing `cron` key rather than introduce a new key. Hopefully it wouldn't get too messy. Totally take a swing at it and I'm happy to help finish it up if it gets hairy.

**Comment by bensheldon** [comment]
I guess I'm imagining that the proc could return a string, in which case it would be passed to fugit, or a time/datetime in which case it would be used. I like your design of passing in `last_run_at`.

**Comment by DanielHeath** [comment]
My reasoning for a new description key was that it can't necessarily be derived from the time proc.

Happy to re-use the `cron` key for the proc argument.

**Comment by bensheldon** [comment]
Implemented in #1051
